content/post/setup-kubernetes-raspberry-pi-cluster.md

```diff
@@ -70,106 +70,121 @@ The installation requries root privileges. Retrieve them by
 
-```
-kubectl expose deployment hypriot --port 80 --type NodePort
+```console
+$ kubectl expose deployment hypriot --port 80 --type NodePort
 ```
 
-In contrast to Docker, K8s itself does not provide an option to define a specific port that listens for a service. According to Lucas is this a design decision, since routing of incoming requests to the front server of the cluster should be handled by a loadbalancer or a webserver. Also see [ingress](http://kubernetes.io/docs/user-guide/ingress/), an extension of K8s that can be configured to route incoming requests to the correct target.
+In contrast to Docker Swarm, Kubernetes itself does not provide an option to define a _specific port_ that listens for a service on the master node. According to Lucas is this an important design decision; routing of incoming requests should be handled by a third party loadbalancer or a webserver, not the core product. The core Kubernetes should be lean and extensible, and encourage others to build tools on top of it for their specific needs.
 
-K8s assigns ports in the range of 30000-32767 to a deployment (think of 'deployment' as a 'Docker service'). To get the port of the service that we just created, run:
+Often Kubernetes offers one possible implementation for a given task, like `kubeadm`. Kubernetes can be set up on so many other custom ways, but since most of the users want it simple, it makes sense to build a opinionated tool that has sane defaults and works for the maximum amount of people.
 
-```
-kubectl get svc hypriot
-```
+In this case, regarding loadbalancers in front of a cluster, there is [the Ingress API object](http://kubernetes.io/docs/user-guide/ingress/) and some sample [Ingress Controllers](https://github.com/kubernetes/ingress). Ingress is a built-in way of exposing Services to the outside world via an Ingress Controller that anyone can build. An Ingress rule defines how traffic should flow from the node the Ingress controller runs on to Services inside of the cluster. Anyway, the bare-metal Ingress Controller that is located in the Kubernetes repositories is unfortunately not ported to ARM yet, although it's on Lucas' list of things to do in 2017 :)
 
-In the output, look at the column **PORT(S)**. The second number is the port at which you can access the service from outside the cluster like at **http://\<ip of the a node>:\<port>**, like http://192.168.0.34:32444 .
+Back to the NodePort feature of a Kubernetes Service. When a Service is of type NodePort, Kubernetes assigns the Service a random port in the range of 30000-32767 **on every node in the cluster**. To get the NodePort of the service that we just created, run:
 
-//TODO add screenshot of browser
+```console
+$ kubectl get service hypriot -o template --template="{{ (index .spec.ports 0).nodePort }}"
+```
 
+You may then access your service on any given node on that port, given that all your `hypriot` Pods are running.
```

Comment:
Let's provide more precise instructions here and say something like: "You may then open your browser and access the service on any given node on that port. For instance at **http://192.168.0.5:32211**."

I would delete the sentence "Given that all your hypriot..." completely since the case that the service is not running is handled in the next instruction.